### ISSUE TITLE
Error importing a filename with a unicode character stopped import of rest of library

### ISSUE THREAD

**Issue by seanblanchfield** [music-assistant/hass-music-assistant#542]
### What version of Music Assistant has the issue?

2022.6.6

### The problem

I have a music library mounted from my NAS via SMB to `/media/music`. I have "filesystem provider" enabled for this path in the music assistant integration configuration. The integration successfully imported 38 albums from the library, but then encountered a `FileNotFoundError` on one of the files, which stopped the entire import. When I fixed the problem with the filename, the rest of the import proceeded okay. I expect you want the `filesystem` provider to be more resilient than this.

### How to reproduce

Place a file named "test∕file.flac" (note the unicode division slash character) in a smb mountpoint that is scanned by the filesystem provider. Wait for it to do a scan and you should see the crash error in the HA core log. This might also be reproducible simply by placing a file with this name, or possibly any unicode character, in a regular local directory read by the filesystem provider. 

### Relevant log output

```
Logger: music_assistant
Source: /usr/local/lib/python3.9/site-packages/music_assistant/mass.py:244
First occurred: 26 June 2022, 15:55:56 (7 occurrences)
Last logged: 09:20:53

Job [Library sync for provider Filesystem] failed with error [Errno 2] No such file or directory: '/media/music/ALBUM/The?Song.flac'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/music_assistant/music_providers/filesystem.py", line 158, in sync_library
    stat = await asyncio.get_running_loop().run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/media/music/ALBUM/The?Song.flac'
```
(I have replaced the original path with `ALBUM/The?Song.flac` above. The important part is the "?").

### Additional information

The problem with the file was that it included a unicode character in its filename that confused either SMB or *hass-music-assistant's* `filesystem` provider. On the filesystem the "?" in the filename is actually the unicode division slash U+2215 "∕ ".  When the `filesystem` provider tries to read the file, it seems to have replaced that unicode character with a "?", causing a `FileNotFoundError`. This error seems not to be handled, and seems to have crashed the entire import. This seems consistent with python code that calls `.encode("ascii", "replace")` on the filename, which would replace unicode with the ascii "?", and subsequently not find the file.

I got around this by finding the offending file and removing the unicode character. This allowed the rest of the library to import. I did not test if the same error is reproducible with a local file containing this unicode character, but I suppose it probably is and would therefore be a bug you would like to fix (even if that just means logging a warning and skipping the problem file).

### What version of Home Assistant Core are your running

2022.6.0

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

ODROID

**Comment by erkr** [comment]
Thanks for your issue. We had a similar issue a while ago with an `'` in the name. As we couldn't reproduce, I guess the user had an Unicode in the name as well.

**Comment by marcelveldt** [comment]
This error is induced by Python builtin libraries, not Music Assistant, completely crashing. 
I am going to catch the exception and log it so the scan does not halt completely but I do like to note that probably SMB is to blame here as the only time it happens is on smb mounted locations.

I can not reproduce the issue on my setup and tried several unicode characters in the filename.

Anyways, I'll fix the bug in MA to not choke on the FileNotFoundError so you can at least spot what's going on but the real fix is probably somewhere else.

**Comment by erkr** [comment]
I agree, not chocking is the only thing needed on MA side to close the issue👍

**Comment by marcelveldt** [comment]
should be fixed in 2022.6.7, if not please reopen, thanks!
